[Merged by Bors] - currency docs (#1305)
ComposableFi/composable: frame/currency-factory/README.md

# Overview

This pallet creates new sovereign(local consensus) currencies(assets) from thin air.

Each currency falls into one of the following categories:

- new currency issued by governance
- currency produced as consequences of execution of protocol and depended on some base currency(ies), usually named LP
- currency mapping the remote location of the asset to local identifiers, foreign currency.

Some well-known identifiers are baked into the runtime during compilation, some are added as part of genesis chain specification and some are created in runtime.
Compilation time currencies are not registered in this pallet.
  
## Basics

Currency must be identified, so this pallet allows the production of new identifiers.

> A Currency needs an associated amount, associated amounts must be stored on accounts, and accounts must pay ED to prevent spam.
> So each currency must have [ED](../../docs/0002-rent-deposit.md) defined.

Reviewer: ```suggestion
A Currency needs an associated amount stored on accounts. Said accounts must pay ED(Existential Deposit) to prevent spam.
So each currency must have an [ED](../../docs/0002-rent-deposit.md) defined.
```

> In rare edge cases, ED could be zero, but in this case, the amount is locked into a protocol, and users cannot freely transfer amounts.
> In some cases, such amounts could be stored in storage without issuing new currency via this factory.

Reviewer: ```suggestion
In rare edge cases, the ED could be zero, but in this case, the asset is locked into a protocol, and users cannot freely transfer.
 We could sometimes store funds without issuing new currency via this factory.
```

All local currencies are normalized to 12 decimals.

## Metadata

> In some cases, governance may add metadata to make currency recognizable, such as:

Reviewer: ```suggestion
In some cases, governance may add metadata to make a currency recognizable, such as:
```

# Overview

> This pallet creates new sovereign(local consensus) currencies(assets) from thin air.

Reviewer 2: ```suggestion
This pallet creates new sovereign (local consensus) currencies (assets) from thin air.
```

Each currency falls into one of the following categories:

- new currency issued by governance
- currency produced as consequences of a protocol execution that depends on one or more base currencies, usually named LP
- currency mapping the remote location of an asset to local identifiers associated with the foreign currency.

Adding well-known identifiers for assets occurs under the following conditions:

- baked into the runtime during compilation
- added as part of genesis chain specification
- created in runtime

> Not registered in this pallet are compilation time currencies.

Reviewer 2: ```suggestion
Compile-time currencies are not registered in this pallet.
```

  
## Basics

> We must identify currency. This pallet allows the production of new identifiers to fill that requirement.

Reviewer 2: ```suggestion
In order to be used, We must be able to uniquely identify currencies. This pallet allows the production of new identifiers (ids) to fill that requirement.
```

> A Currency needs an associated amount stored on accounts. Said accounts must pay ED(Existential Deposit) to prevent spam.
> So each currency must have an [ED](../../docs/0002-rent-deposit.md) defined.

Reviewer 2: ```suggestion
A Currency needs an associated amount stored on accounts. To prevent spam, accounts are required to pay an ED (Existential Deposit). As such, each currency must have an [ED](../../docs/0002-rent-deposit.md) defined.
```

Reviewer 2: Unsure if "A Currency needs an associated amount stored on accounts." can be removed

> In rare edge cases, the ED could be zero, but in this case, the asset is locked into a protocol, and users cannot freely transfer.
>  We could sometimes store funds without issuing new currency via this factory.

Reviewer 2: ```suggestion
In rare edge cases, the ED could be zero, but in this case, the asset is locked into a protocol, and users cannot freely transfer.

We could sometimes store funds without issuing new currency via this factory.
```

Reviewer 2: Is there an example of one of these edge cases?

All local currencies are normalized to 12 decimals.

## Metadata

In some cases, governance may add metadata to make a currency recognizable, such as:

- name

> - `symbol` Currency may have a human-readable symbol. For example, `XBTC`. This metadata targets for governance to prevent spam and fishing.

Reviewer 2: ```suggestion
- `symbol`: A currency may have a human-readable symbol. For example, `XBTC`. This metadata targets for governance to prevent spam and fishing.
```

Reviewer 2: Not sure what is meant by "This metadata targets for governance [...]"

## Foreign integration

[AssetsRegistry](../assets-registry/README.md) uses this pallet to integrate other decimals and out-of-consensus locations.

Each foreign currency **must** have an entry in AssetsRegistry too.

## Approved governance currencies

Allows governance to approve or revoke currencies to participate in democracy.

## Ranges

> This pallet proves the ability to split identifiers into some nominal numeric ranges to help identify assets easier and prevents conflicts.

Reviewer 2: ```suggestion
This pallet provides the ability to split identifiers into some nominal numeric ranges to help identify assets easier and prevent conflicts.
```
